packages/data-point/lib/reducer/resolve.js

```diff
@@ -41,17 +45,17 @@ module.exports.getReducerFunction = getReducerFunction
 /**
  * apply a Reducer to an accumulator
  * @param {Object} manager
- * @param {Function} resolveTransform
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
  * @returns {Promise<Accumulator>}
  */
-function resolve (manager, resolveTransform, accumulator, reducer) {
-  const reducerFunction = getReducerFunction(
-    manager,
-    resolveTransform,
-    reducer.type
-  )
+function resolve (manager, accumulator, reducer) {
+  if (!reducer) {
+    return Promise.resolve(accumulator)
+  }
```

Comment:
this conditional is here because the entity resolve function does not check if lifecycle functions are defined before trying to resolve them:

https://github.com/raingerber/data-point/blob/rename-transform-expression/packages/data-point/lib/entity-types/base-entity/resolve.js#L140